### ISSUE TITLE
Add ability to handle snapshots to vsphere_guest module

### ISSUE THREAD

**Issue by ronator** [ansible/ansible-modules-core#4942]
Original issue was https://github.com/ansible/ansible-modules-core/issues/2671. I used a wrong name in that ticket, I meant vsphere_guest. I was suggested by ansibot to re-open it here.

ansibot commented 7 minutes ago
@ronator: Thanks for taking the time to open this issue, but it was filed under the wrong github repository. Please create a new issue at https://github.com/ansible/ansible-modules-core/issues
##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

vsphere_guest
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### OS / ENVIRONMENT

"N/A"
##### SUMMARY

The vmware_guest module lacks the ability to handle (list/create/remove) **snapshots** of vmware guests. This would be a killer feature! It would also allow more automization of the 'snapshot_to_clone' option, because for now using this option, snapshot must already be taken in vCenter.

**Comment by ansibot** [comment]
@chrrrles, @rhoop, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@chrrrles, @rhoop, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Abdul-Anshad-A** [comment]
@ronator I have raised a pull request to handle snapshot operations with vsphere_guest and its pending on maintainer's review. Meanwhile, can you help me out with testing the changes ? You will find more instruction on pull request #5390 

https://github.com/Abdul-Anshad-A/ansible-modules-core/blob/snapshot_ops/cloud/vmware/vsphere_guest.py

**Comment by ronator** [comment]
Sorry, I missed your message. I will read it and see if can help you. Embarrassingly, I do not know how to test it. I am relatively new to github. I cloned your branch "snapshots_ops" and know I am stuck :(
Anyway I have to say :+1: for the effort.

**Comment by Abdul-Anshad-A** [comment]
@ronator No issues, Let me try to explain the testing process as simple as possible.

### 1)Install from source & setup environment

```
$ git clone git://github.com/ansible/ansible.git --recursive
$ cd ./ansible
$ source ./hacking/env-setup
```

### 2) Replace vsphere_guest module with the new modified one

`cp -f <path_to_vsphere_guest.py_cloned_from_my_snapshot_ops_branch> ./lib/ansible/modules/core/cloud/vmware/vsphere_guest.py`

### 3)Create play books accordingly as instructed on pull request #5390

Example:

```
vsphere_guest:
    vcenter_hostname: <hostname>
    username: <username>
    password: <password>
    validate_certs: False
    guest: <vm_name>
    snapshot_op:
        op_type: list_all
```

NOTE - Please, test it on your non-production environment.

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29640
